site/app/libraries/GradingQueue.php

```diff
@@ -209,4 +211,220 @@ public function getGradingCount() {
+            $elapsed_time .= str_pad(floor($job_file["elapsed_time"] / (60 * 60)), 2, "0", STR_PAD_LEFT) . ":";
+            $job_file["elapsed_time"] = $job_file["elapsed_time"] % (60 * 60);
+            // Format the string with the minutes and seconds elapsed
+            $elapsed_time .= str_pad(floor($job_file["elapsed_time"] / 60), 2, "0", STR_PAD_LEFT) . ":"
+                . str_pad($job_file["elapsed_time"] % 60, 2, "0", STR_PAD_LEFT);
+            $ongoing_job_info[$machine][] = [
+                "semester" => $file_segments[0],
+                "course" => $file_segments[1],
+                "gradeable_id" => $file_segments[2],
+                "user_id" => $file_segments[3],
+                "regrade" => $job_file["regrade"],
+                "elapsed_time" => $elapsed_time,
+                "error" => $job_file["error"],
+                "stale" => $job_file["stale"]
+            ];
+        }
+
+        // These are files in the grading directory but does not have a GRADING_ file, so grading has not started yet
+        foreach ($not_yet_started as $file) {
+            $path = FileUtils::joinPaths($this->grading_path, $file);
+            $this->updateDetailedQueueCount($path, false, $epoch_time, $queue_counts, $capability_queue_counts, $machine_grading_counts, $open_autograding_workers_json, $course_info);
+        }
+
+        return [
+            "machine_grading_counts" => $machine_grading_counts,
+            "capability_queue_counts" => $capability_queue_counts,
+            "queue_counts" => $queue_counts,
+            "num_autograding_workers" => $workers,
+            "course_info" => $course_info,
+            "ongoing_job_info" => $ongoing_job_info
+        ];
+    }
+
+    // Helper function used to interpret the job files and update the count variables appropriately
+    private function updateDetailedQueueCount(string $queue_or_grading_file, bool $is_grading, int $epoch_time, array &$detailed_queue_counts, array &$capability_queue_counts, array &$machine_grading_counts, array $open_autograding_workers_json, array &$course_info): array {
+        $stale = false;
+        $error = "";
+        $regrade = false;
+
+        $elapsed_time = -1;
+
+        $job_file = basename($queue_or_grading_file);
+
+        try {
+            $entry = new QueueItem($queue_or_grading_file, $epoch_time, $is_grading);
+            $regrade = $entry->isRegrade();
+            if ($regrade) {
+                if ($is_grading) {
+                    $detailed_queue_counts["regrade_ongoing"] += 1;
+                }
+                else {
+                    $detailed_queue_counts["regrade"] += 1;
+                }
+            }
+            else {
+                if ($is_grading) {
+                    $detailed_queue_counts["interactive_ongoing"] += 1;
+                }
+                else {
+                    $detailed_queue_counts["interactive"] += 1;
+                }
+            }
+            $capability = "default";
```

Comment:
Can we be certain that there will always be a capability named "default"?  (I'm not too familiar with how this works yet)

Reply:
Default should exist from the submitty setup script, it is possible that someone get rid of it if they go into the system itself however.